test/libsolidity/SyntaxTest.cpp

```diff
@@ -46,16 +46,22 @@ SyntaxTest::SyntaxTest(
 	CommonSyntaxTest(_filename, _evmVersion),
 	m_minSeverity(_minSeverity)
 {
-	static std::set<std::string> const compileViaYulAllowedValues{"true", "false"};
-
-	auto const eofEnabled = solidity::test::CommonOptions::get().eofVersion().has_value();
+	m_compileViaYul = m_reader.enumSetting<CompileViaYul>(
+		"compileViaYul",
+		{
+			{toString(CompileViaYul::False), CompileViaYul::False},
+			{toString(CompileViaYul::True), CompileViaYul::True},
+			{toString(CompileViaYul::Also), CompileViaYul::Also},
+			{toString(CompileViaYul::OnlyOnEOF), CompileViaYul::OnlyOnEOF}
+		},
+		toString(CompileViaYul::OnlyOnEOF)
+	);
 
-	m_compileViaYul = m_reader.stringSetting("compileViaYul", eofEnabled ? "true" : "false");
-	if (!util::contains(compileViaYulAllowedValues, m_compileViaYul))
-		BOOST_THROW_EXCEPTION(std::runtime_error("Invalid compileViaYul value: " + m_compileViaYul + "."));
+	solUnimplementedAssert(m_compileViaYul != CompileViaYul::Also,
+		"'compileViaYul: also' is not yet supported in syntax tests.");
 
-	if (m_compileViaYul == "false" && eofEnabled)
-		m_shouldRun = false;
+	if (bytecodeFormat().contains(BytecodeFormat::EOFv1) && m_compileViaYul == CompileViaYul::False)
+		BOOST_THROW_EXCEPTION(std::runtime_error("Compilation to EOF requires using Yul IR"));
```

Comment:
Still needs an assert.

While at it, let's also make the message a bit clearer. We should mention which settings it's about because the error won't point at a specific line in the test.

```suggestion
	if (m_compileViaYul == CompileViaYul::False)
	{
		soltestAssert(!eofEnabled);
		if (bytecodeFormat().contains(BytecodeFormat::EOFv1))
			BOOST_THROW_EXCEPTION(std::runtime_error(
				"Conflicting test settings: 'bytecodeFormat' includes EOF, which requires compilation via IR, but 'compileViaYul' is set to false."
			));
	}
```